typescript/packages/wallets/crossmint/src/custodial.ts

```diff
@@ -38,8 +38,12 @@ function getLocator(params: CustodialOptions): string | number {
     return `userId:${params.userId}:solana-custodial-wallet`;
 }
 
+export interface CrossmintSolanaCustodialWalletClient extends SolanaWalletClient {
+    tokenBalanceOf: (currencies: SupportedSolanaCurrency[], address: string) => Promise<GetWalletBalanceResponse>
```

Comment:
No need to add it here.

The wallet interface just takes care of getting the native token balance.

This would only be used if a plugin would specifically look for `CrossmintSolanaCustodialWalletClient` and use that function

Reply:
hmm this is the case today. But it's really just calling an endpoint, doesn't need anything from the wallet client.

That endpoint could evolve into allowing fetching the balance for any wallet too and then you would need to remove the logic from the wallet client.

I would implement it as the `faucet` plugin and then just add an error that this only works for crossmint wallets for now.